### PR TITLE
[MM-33262] db/migrations: add publicchannels migration

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary

🕺 `publicchannels` migration is here! (This step requires `channels` table to exist. So this is blocked by [MM-33061](https://mattermost.atlassian.net/browse/MM-33061))

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-33262

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: db/migrations/mysql/000031_create_public_channels.down.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+DROP TABLE IF EXISTS PublicChannels;
```
Comment (agnivade) [inline db/migrations/mysql/000031_create_public_channels.down.sql:1]: We aren't dropping the indexes here like in the pg case.

File: db/migrations/mysql/000031_create_public_channels.up.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    KEY idx_publicchannels_name (Name),
+    KEY idx_publicchannels_delete_at (DeleteAt),
+    KEY idx_publicchannels_search_txt (Name, DisplayName, Purpose)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+CREATE PROCEDURE MigratePC()
+BEGIN
+IF(NOT EXISTS (
+		SELECT
+			1 FROM PublicChannels)) THEN
```
Comment (agnivade) [inline db/migrations/mysql/000031_create_public_channels.up.sql:21]: Nit: would look good if all of these were in a single line.

File: db/migrations/postgres/000031_create_public_channels.up.sql [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+CREATE INDEX IF NOT EXISTS idx_publicchannels_name_lower ON publicchannels (lower(name));
+CREATE INDEX IF NOT EXISTS idx_publicchannels_displayname_lower ON publicchannels (lower(displayname));
+CREATE INDEX IF NOT EXISTS idx_publicchannels_search_txt ON publicchannels (name, displayname, purpose);
+
+DO $$
+	<< migratepc >>
+BEGIN
+	IF(NOT EXISTS (
+		SELECT
+			1 FROM publicchannels)) THEN
```
Comment (agnivade) [inline db/migrations/postgres/000031_create_public_channels.up.sql:24]: Same as above.

File: db/migrations/postgres/000031_create_public_channels.up.sql [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+CREATE INDEX IF NOT EXISTS idx_publicchannels_team_id ON publicchannels (teamid);
+CREATE INDEX IF NOT EXISTS idx_publicchannels_name ON publicchannels (name);
+CREATE INDEX IF NOT EXISTS idx_publicchannels_delete_at ON publicchannels (deleteat);
+CREATE INDEX IF NOT EXISTS idx_publicchannels_name_lower ON publicchannels (lower(name));
+CREATE INDEX IF NOT EXISTS idx_publicchannels_displayname_lower ON publicchannels (lower(displayname));
+CREATE INDEX IF NOT EXISTS idx_publicchannels_search_txt ON publicchannels (name, displayname, purpose);
+
+DO $$
+	<< migratepc >>
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply db/migrations/postgres/000031_create_public_channels.up.sql]: Hmm, what do you mean by verify? it was in the migrations file `upgrade.go`.

[the remark this answers is not in the document]
Comment (agnivade) [reply db/migrations/postgres/000031_create_public_channels.up.sql]: I mean whether this syntax is correct or not. Yes, the migration needs to be run, but from the Postgres docs I linked above, I don't see the `<< migratepc >>` part required. So just wanted to ensure whether everything is fine or not.

[the remark this answers is not in the document]
Comment (isacikgoz) [reply db/migrations/postgres/000031_create_public_channels.up.sql]: Oh, right. Yeah I this was happened to be in other implementations and then I used it as well. We can remove it to keep things simple.

**Review by agnivade — APPROVED** [review]

**Review by mgdelacroix — APPROVED** [review]
LGTM!! :+1:

**Pull request merged**